### ISSUE TITLE
Another name conflict with MemoryMappedFile tests

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#722]
Just hit this running build locally, fully sync'd to master...

```
Discovering: MemoryMappedFile.Tests
Discovered:  MemoryMappedFile.Tests
Starting:    MemoryMappedFile.Tests
ERROR, Loc411: Unexpected exception, System.IO.IOException: Cannot create a file when that file already exists.

   at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateCore(SafeFileHandle fileHandle, String mapName, HandleInheritability inheritability, MemoryMappedFil
eAccess access, MemoryMappedFileOptions options, Int64 capacity) in c:\Users\stoub\Source\Repos\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMa
ppedFiles\MemoryMappedFile.Windows.cs:line 40
   at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateNew(String mapName, Int64 capacity, MemoryMappedFileAccess access, MemoryMappedFileOptions options,
HandleInheritability inheritability) in c:\Users\stoub\Source\Repos\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.c
s:line 308
   at CreateNew.VerifyCreateNew(String strLoc, String mapName, Int64 capacity, MemoryMappedFileAccess access, MemoryMappedFileOptions options, HandleInherita
bility inheritability) in c:\Users\stoub\Source\Repos\corefx\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateNew.cs:line 325
Fail: iCountErrors==1
   CreateNew.CreateNewTestCases [FAIL]
      One or more test cases failed.
      Stack Trace:
         c:\Users\stoub\Source\Repos\corefx\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateNew.cs(28,0): at CreateNew.CreateNewTestCases()
Finished:    MemoryMappedFile.Tests

=== TEST EXECUTION SUMMARY ===
```

**Comment by stephentoub** [comment]
This happened again in a recent CI run, but in a different test (maybe the two are related?)

MemoryMappedFile.Tests -> d:\j\workspace\dotnet_corefx_windows_release_prtest\bin\Windows_NT\Release\MemoryMappedFile.Tests\MemoryMappedFile.Tests.dll
  xUnit.net console test runner (32-bit .NET Core)
  Copyright (C) 2014 Outercurve Foundation.
  
  Discovering: MemoryMappedFile.Tests
  Discovered:  MemoryMappedFile.Tests
  Starting:    MemoryMappedFile.Tests
  ERROR, Loc214: Unexpected exception, System.IO.IOException: Cannot create a file when that file already exists.
  
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateCore(SafeFileHandle fileHandle, String mapName, HandleInheritability inheritability, MemoryMappedFileAccess access, MemoryMappedFileOptions options, Int64 capacity) in d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.Windows.cs:line 24
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity, MemoryMappedFileAccess access) in d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 180
     at CreateFromFile.VerifyCreateFromFile(String strLoc, String fileName, FileMode fileMode, String mapName) in d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateFromFile.cs:line 704
  Fail: iCountErrors==1
     CreateFromFile.CreateFromFileTestCases [FAIL]
        One or more test cases failed.
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateFromFile.cs(28,0): at CreateFromFile.CreateFromFileTestCases()
  Finished:    MemoryMappedFile.Tests
  
  === TEST EXECUTION SUMMARY ===
     MemoryMappedFile.Tests  Total: 7, Errors: 0, Failed: 1, Skipped: 0, Time: 0.338s

**Comment by joshfree** [comment]
One more


http://dotnet-ci.cloudapp.net/job/dotnet_corefx_windows_debug/38/console 

=== TEST EXECUTION SUMMARY ===
     System.IO.FileSystem.Tests  Total: 338, Errors: 0, Failed: 0, Skipped: 0, Time: 0.583s
  System.IO.MemoryMappedFiles -> d:\j\workspace\dotnet_corefx_windows_debug\bin\Windows_NT\Debug\System.IO.MemoryMappedFiles\System.IO.MemoryMappedFiles.dll
  MemoryMappedFile.Tests -> d:\j\workspace\dotnet_corefx_windows_debug\bin\Windows_NT\Debug\MemoryMappedFile.Tests\MemoryMappedFile.Tests.dll
  xUnit.net console test runner (32-bit .NET Core)
  Copyright (C) 2014 Outercurve Foundation.
  
  Discovering: MemoryMappedFile.Tests
  Discovered:  MemoryMappedFile.Tests
  Starting:    MemoryMappedFile.Tests
  ERROR, Loc214: Unexpected exception, System.IO.IOException: Cannot create a file when that file already exists.
  
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateCore(SafeFileHandle fileHandle, String mapName, HandleInheritability inheritability, MemoryMappedFileAccess access, MemoryMappedFileOptions options, Int64 capacity) in d:\j\workspace\dotnet_corefx_windows_debug\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.Windows.cs:line 40
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity, MemoryMappedFileAccess access) in d:\j\workspace\dotnet_corefx_windows_debug\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 180
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName) in d:\j\workspace\dotnet_corefx_windows_debug\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 102
     at CreateFromFile.VerifyCreateFromFile(String strLoc, String fileName, FileMode fileMode, String mapName) in d:\j\workspace\dotnet_corefx_windows_debug\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateFromFile.cs:line 707
  Fail: iCountErrors==1
     CreateFromFile.CreateFromFileTestCases [FAIL]
        One or more test cases failed.
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_debug\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateFromFile.cs(28,0): at CreateFromFile.CreateFromFileTestCases()
  Finished:    MemoryMappedFile.Tests

**Comment by joshfree** [comment]
@FiveTimesTheFun can you look into fixing this test case today as you switch over to working on more of System.IO?

**Comment by FiveTimesTheFun** [comment]
Thanks @joshfree, will do.

**Comment by FiveTimesTheFun** [comment]
Fixed with a31691d65faf9028483e4f9aabe725286d89b0e6
